File: Day1PrefixSum/PrefixSum.py
# print("Test!")
def runningSum(lst):
    # start at 1
    # go to the lenght of the list
    for i in range(1, len(lst)):
        # add the next number to the last number
        # save as next number
        lst[i] = lst[i] + lst[i-1]
    # return updated list
    return lst
print(runningSum([1,2,3,4]))

def pivotIndex(lst):
    # initialize right and left sums
    right=0 
    left=0
    # loop through to get right sum
    for e in lst:
        right = right + e
    print(right)
    # if left sum equals the right sum minus the element
    # return the index
    for i in range(len(lst)):
        if left == (right - lst[i]):
            return i
        # else
        # add elent to left
        # subtract element from right
        left = left + lst[i]
        right = right - lst[i]
    
    return -1

print(pivotIndex([1,7,3,6,5,6]))
        

File: Day1PrefixSum/PrefixSum.java
public class PrefixSum{
    public static int[] runningSum(int[] arr){
        for (int i = 1; i < arr.length; i++) {
            // add the next number to the previous number
            // save as the next number
            arr[i] += arr[i - 1];
            /*  2 : 1
            3 : 3
            4 : 6 */
        }
        return arr;
    }

    public static int pivotIndex(int[] arr){
        // initialize right and left sums
        int right = 0, left = 0;
        // get right sum
        for (int i: arr) right += i;
        // get left sum
        for (int i = 0; i < arr.length; ++i) {
            // if left sum is the same as right sum minus element
            // return index
            if (left == (right - arr[i])) return i;
            // else
            // add element to left sum
            // subtract element from right sum
            left += arr[i];
            right -= arr[i];
        }
        // if nothing return -1
        return -1;
    }
    public static void main(String[] args) {
        // System.out.println("Test java");
        runningSum(new int[] {1,2,3,4});
        System.out.println( pivotIndex(new int[] {1,7,3,6,5,6}) );
    }
}